libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c

```diff
@@ -389,6 +389,64 @@ Socket_t SOCKETS_Socket( int32_t lDomain,
 
 /*-----------------------------------------------------------*/
 
+int32_t SOCKETS_Bind( Socket_t xSocket,
+                      SocketsSockaddr_t * pxAddress,
+                      Socklen_t xAddressLength )
+{
+    ss_ctx_t * ctx;
+    int32_t ret;
+    struct sockaddr_in sa_addr = { 0 };
+
+    if( SOCKETS_INVALID_SOCKET == xSocket )
+    {
+        configPRINTF( ( "TCP socket Invalid\n" ) );
+        return SOCKETS_EINVAL;
+    }
+
+    if( pxAddress == NULL )
+    {
+        configPRINTF( ( "TCP socket Invalid Address\n" ) );
+        return SOCKETS_EINVAL;
+    }
+
+    ctx = ( ss_ctx_t * ) xSocket;
```

Comment:
Should there be a NULL check against `ctx`?

Reply:
@aggarw13 

This is already checked in line 400

 if( SOCKETS_INVALID_SOCKET == xSocket )
  {
        configPRINTF( ( "TCP socket Invalid\n" ) );
        return SOCKETS_EINVAL;
  }
so it is not needed.

Thanks
Balaji.

Reply:
Hi Balaji, the check against `SOCKETS_INVALID_SOCKET` is for a value of  `-1` whereas a `NULL` pointer value would resolve to to `0`. Thus, I recommend checking against `NULL` to avoid any SegFault issue from derefencing an invalid `ss_ctx_t *` type value

Reply:
@aggarw13 

I have added the null check for socket context